Polygon_mesh_processing/doc/Polygon_mesh_processing/Polygon_mesh_processing.txt

```diff
@@ -1416,6 +1416,84 @@ and the number of surface patches that are separated by these edges.
+The core algorithm is ICP, iterative closest point. In each iteration, the method identifies vertex or point pairs between the template, initially a copy of the source, and the target and subsequently calculates the transformations which minimize the distance between them. The template is updated after each iteration and converges towards the target.
+
+\f{equation}{
+\mathbf{E_{reg}} = w_1\mathbf{E_{match}} + w_2\mathbf{E_{point\_to\_plane}} + w_3\mathbf{E_{arap}},
+  \label{eq:e_reg}
+\f}
+
+with the single energy terms:
+
+\f{equation}{
+\mathbf{E_{match}} = \sum_{\mathbf{v}_i, \mathbf{\tilde{v}}_i \in P} \left\|\mathbf{\tilde{v}}_i  - (\mathbf{R}_i\mathbf{v}_i + \mathbf{t_i})\right\|^2_2,
+  \label{eq:e_match}
+\f}
+
+\f{equation}{
+\mathbf{E_{point\_to\_plane}} = \sum_{\mathbf{v}_i, \mathbf{\tilde{v}}_i \in P} \left\|(\mathbf{\tilde{v}}_i  - (\mathbf{R}_i\mathbf{v}_i + \mathbf{t_i}))\mathbf{\tilde{n}}_i\right\|^2_2,
+  \label{eq:e_plane}
+\f}
+
+\f{equation}{
+\mathbf{E_{arap}} = 
+\sum_{\mathbf{v}_i \in M}
+\sum_{\mathbf{v}_j \in N(\mathbf{v}_i)}
+\left\| (\mathbf{R}_i\mathbf{v}_i + \mathbf{t}_i - \mathbf{R}_j\mathbf{v}_j + \mathbf{t}_j) - \mathbf{R}_i(\mathbf{v}_i - \mathbf{v}_j) \right\|^2_2,
+  \label{eq:e_arap}
+\f}
+
+where:
+- \f$w_1, w_2\f$ and \f$w_3\f$ denote the weights for the energy terms.
+- \f$\mathbf{P}\f$ is the set of vertex/point pairs with \f$\mathbf{v}_i \in\f$ template and \f$\mathbf{\tilde{v}}_i \in\f$ target. It is recreated in each iteration.
+- \f$\mathbf{R}_i\f$ is a \f$ 3 \times 3 \f$ rotation matrix for source vertex \f$\mathbf{v}_i\f$.
+- \f$\mathbf{t}_i\f$ is the translation vector for source vertex \f$\mathbf{v}_i\f$.
+- \f$\mathbf{M}\f$ is the template mesh (initially equal to the source mesh).
+- \f$N(\mathbf{v}_i)\f$ denotes the set of vertices adjacent to \f$\mathbf{v}_i\f$.
+
+The \f$\mathbf{E_{match}}\f$ energy penalizes the distance between point pairs \f$\mathbf{v}_i\f$ and \f$\mathbf{\tilde{v}}_i\f$. The point pairs are restablished after each iteration. In addition, the method can take fixed point pairs of correspondences between source and target. This greatly improves the quality of the registration and is generally required for non-simple meshes.
+
+The \f$\mathbf{E_{point\_to\_plane}}\f$ energy, point to plane, is similar to \f$\mathbf{E_{match}}\f$, but instead penalizes the distance to the plane given by the target vertex and its normal. This energy compensates for different discretizations, i.e., for source and target mesh that were not created by deforming one mesh into the other.
+
+The \f$\mathbf{E_{arap}}\f$ energy, as-rigid-as-possible, controls the rigidness of the transformation. It measures the deformation of the edges connected to each vertex that results from applying an individual transformation to each vertex of the source mesh.
```

Comment:
I don't know if rigidness is ok or if it should be rigidity. We should ask Roberto